<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitXMLReporter -->
<testsuite ignored="0" hostname="Neena" failures="2" tests="8" name="subcategory" time="42.97" errors="0" timestamp="2024-11-20T19:54:21 IST">
  <testcase classname="com.sevenrmartsupermarket.tests.SubcategoryTest" name="verifyAddingNewSubCategory" time="0.0">
    <failure type="org.testng.TestNGException" message="
Method public void com.sevenrmartsupermarket.tests.SubcategoryTest.verifyAddingNewSubCategory() requires a @DataProvider named : newsubcategory in class com.sevenrmartsupermarket.constants.Constants">
      <![CDATA[org.testng.TestNGException: 
Method public void com.sevenrmartsupermarket.tests.SubcategoryTest.verifyAddingNewSubCategory() requires a @DataProvider named : newsubcategory in class com.sevenrmartsupermarket.constants.Constants
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1570)
... Removed 14 stack frames]]>
    </failure>
  </testcase> <!-- verifyAddingNewSubCategory -->
  <testcase classname="com.sevenrmartsupermarket.tests.SubcategoryTest" name="verifyInvalidSearch" time="4.267"/>
  <testcase classname="com.sevenrmartsupermarket.tests.SubcategoryTest" name="verifyInvalidSearchUsingDataProvider" time="2.532"/>
  <testcase classname="com.sevenrmartsupermarket.tests.SubcategoryTest" name="verifyInvalidSearchUsingDataProvider" time="2.384"/>
  <testcase classname="com.sevenrmartsupermarket.tests.SubcategoryTest" name="verifyInvalidSearchUsingDataProvider" time="3.212"/>
  <testcase classname="com.sevenrmartsupermarket.tests.SubcategoryTest" name="verifyResetButtonIsenabledOrNot" time="1.687"/>
  <testcase classname="com.sevenrmartsupermarket.tests.SubcategoryTest" name="verifySearchResult" time="2.964">
    <failure type="java.lang.IndexOutOfBoundsException" message="toIndex = 2">
      <![CDATA[java.lang.IndexOutOfBoundsException: toIndex = 2
at java.base/java.util.AbstractList.subListRangeCheck(AbstractList.java:509)
at java.base/java.util.ArrayList.subList(ArrayList.java:1191)
at com.sevenrmartsupermarket.tests.SubcategoryTest.verifySearchResult(SubcategoryTest.java:45)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1570)
... Removed 15 stack frames]]>
    </failure>
  </testcase> <!-- verifySearchResult -->
  <testcase classname="com.sevenrmartsupermarket.tests.SubcategoryTest" name="verifySubcategoryListUsingSearchbutton" time="2.737"/>
</testsuite> <!-- subcategory -->
